target_snowflake/__init__.py

```diff
@@ -257,6 +276,26 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
                 else:
                     stream_to_sync[stream] = DbSync(config, o, table_cache, file_format_type)
 
+                if archive_load_files_enabled:
+                    # Determine archive_load_files_primary_column
+                    # 1) Use incremental replication key if defined
+                    # 2) Otherwise use primary key
+                    if False and 'bookmark_properties' in o and len(o['bookmark_properties']) > 0:
```

Comment:
TBD: What's the proper way to identify incremental replication key

Reply:
check [messages-simple-table.json](https://github.com/transferwise/pipelinewise-target-snowflake/blob/ad0d71baa9dee4c47b05869b7ca6c333593f8da3/tests/integration/resources/messages-simple-table.json), maybe this is a better example.

The incremental key is defined in details only in the `STATE` message, but taps are not sending them necessarily always after the `SCHEMA` and before the `RECORD` messages. The order is not defined by the [singer-spec](https://github.com/singer-io/getting-started/blob/master/docs/SPEC.md#state-message)

But taps are copying replication key into the `bookmark_properties`. If it's logical replication then tap-postgres using the `lsn` keyword. 

I think the best what we can do is to get `o['bookmark_properties']` and if the result is one of the column in the schema then we use it as `archive_load_files_primary_column`. If not exists then we use the PK: o['key_properties'][0].

Incremental key is a single value, but primary key can have multiple columns